### DISCUSSION TITLE
Different controller

### DISCUSSION THREAD

**Discussion by iMonZ** [LizardByte/community#323] [Sunshine] [answered]
So Game Controller Are assigned as XBOX 360
So no native features are working with Nintendo switch, Xbox One or DualSense Controller. Is this an issue with Moonlight or Sunshine?
And why is that so?

**Comment by ReenigneArcher** [comment]
Sunshine replicates the behavior of GFE, which emulated an Xbox 360 gamepad. Going forward we can make changes, but we rely on other tools, which currently do not support newer gamepads. We do already support DS4 on Windows though.

**Reply by iMonZ** [reply]
Thanks. Do you know which tools need an update so that DualSense 5 and Nintendo Switch Pro Controller is fully supported on platforms like Linux?
And also if sunshine has the updated support, is an update to the moonlight client needed as well?
Still thanks for the help

**Reply by ReenigneArcher** [reply, marked as answer]
Sunshine supports Xbox One, DS5, and Nintendo Switch Pro on Linux now. Not on Windows though.
